lib/flight_job/models/job.rb

```diff
@@ -332,14 +332,50 @@ def format_estimated_end_time(verbose)
       end
     end
 
-    def serializable_hash
+    def stdout_readable?
+      return false unless stdout_path
+      return false unless File.exists? stdout_path
+      File.stat(stdout_path).readable?
+    end
+
+    def stderr_readable?
+      return false unless stderr_path
+      return false unless File.exists? stderr_path
+      File.stat(stderr_path).readable?
+    end
+
+    def serializable_hash(opts = nil)
+      opts ||= {}
       {
         "id" => id,
         "actual_start_time" => actual_start_time,
         "estimated_start_time" => estimated_start_time,
         "actual_end_time" => actual_end_time,
-        "estimated_end_time" => estimated_end_time
-      }.merge(metadata)
+        "estimated_end_time" => estimated_end_time,
+        # NOTE: The API uses the 'size' attributes as a proxy check to exists/readability
+        #       as well as getting the size. Non-readable stdout/stderr would be
+        #       unusual, and can be ignored
+        "stdout_size" => stdout_readable? ? File.size(stdout_path) : nil,
+        "stderr_size" => stderr_readable? ? File.size(stderr_path) : nil
+      }.merge(metadata).tap do |hash|
+        if opts.fetch(:include, []).include? 'script'
+          hash['script'] = load_script
+        end
+
+        if opts.fetch(:include, []).include? 'result_files'
+          if results_dir && Dir.exist?(results_dir)
+            files =  Dir.glob(File.join(results_dir, '**/*'))
+                        .map { |p| Pathname.new(p) }
+                        .reject(&:directory?)
+                        .reject(&:symlink?)
+                        .select(&:readable?) # These would be unusual and should be rejected
+                        .map { |p| { file: p.to_s, size: p.size } }
```

Comment:
I don't think there is much benefit adding it in.

The `mimeType` is a web concept and can be easily inferred from the file extension. It would be out of place within a CLI application.

The API is already expecting the `file` to be within the `results_dir` and works out the relative path accordingly. Adding it in would be needlessly verbose.

Reply:
My question was more "does the API still include it; or does it need adding here?".  As long as its included in the API response that's fine.  Inferring it from the file extension is also fine.

---

Aside:

There are a number of CLI programs that use mime types; `file` and `xdg-open` spring to mind.  They are not necessarily web concepts and are not necessarily out of place within a CLI application.  Some programs calculate mime types from the file extension whilst others use "magic".  See below for magic.

```
$ ls
test.docx
$ file test.docx 
test.docx: Microsoft Word 2007+
$ file -i test.docx 
test.docx: application/vnd.openxmlformats-officedocument.wordprocessingml.document; charset=binary
$ mv test.docx test.pdf
$ file test.pdf 
test.pdf: Microsoft Word 2007+
$ file -i test.pdf
test.pdf: application/vnd.openxmlformats-officedocument.wordprocessingml.document; charset=binary
```

Reply:
I'm not sure exactly, what you mean by that?

The API does indeed return both the `relative_path` and the `mime_type`. Currently `flight-job` provides the `results_dir` and absolute path to the API.  The API determines the `relative_path` from the `results_dir` and the `mime_type` from the file extension.

I don't see much benefit moving the `ruby` code from API into the CLI; as there is no permissions or speed improvements. It would only increase the amount of data that is being piped between the processes.

---

I still think the _Multipurpose Internet Mail Extensions_ would be out of place in `flight-job`. It doesn't use the MIME type nor provide any special handling to determine it. In this situation, it is very much a web concept.

I believe `file` has some special handling to infer the MIME type from the file's content (as opposed to the extension) however WAGNI. Requiring sensible file extensions should be sufficient for the time being.

Reply:
> The API does indeed return both the relative_path and the mime_type. Currently flight-job provides the results_dir and absolute path to the API. The API determines the relative_path from the results_dir and the mime_type from the file extension.

Excellent.  That's all I wanted to know.

---

`file` uses magic.